Widgets/ResourceSelector.h

```diff
@@ -0,0 +1,30 @@
+#ifndef RESOURCESELECTOR_H
+#define RESOURCESELECTOR_H
+
+#include "Components/QMenuView.h"
+#include "MainWindow.h"
+#include "Models/TreeSortFilterProxyModel.h"
+
+#include <QToolButton>
+
+class ResourceSelector : public QToolButton {
+ public:
+  explicit ResourceSelector(QWidget* parent, TreeNode::TypeCase type = TreeNode::TypeCase::kObject)
```

Comment:
I feel dirty defaulting this to object. This is just so this constructor can serve as the default, yes?

Reply:
Without a default argument the designer's generated code craps out. I can default the type to none if you prefer but I do need a default